src/main/java/johnny/NoDateException.java

```diff
@@ -0,0 +1,15 @@
+package johnny;
+
+public class NoDateException extends Exception{
```

Comment:
```suggestion
public class NoDateException extends Exception {
```
Missing whitespace between Exception and the curly brace.